woocommerce/pinterest-for-woocommerce
# Multichannel Marketing Integration - Basic details and methods to report feed status

### Changes proposed in this Pull Request:



Closes #697.

This PR implements an integration with the new WooCommerce Multichannel Marketing dashboard. I have also done a number of refactoring changes to make the code more reusable. Some minor code style updates are also included in this PR.


### Screenshots:

<img width="716" alt="image" src="https://user-images.githubusercontent.com/73110514/224365177-65032174-c954-4c79-921d-6b82aa3a894f.png">


### Detailed test instructions:


To test this PR it's best if you use the latest development version of WooCommerce (`trunk`).

1. Go to `WooCommerce Settings > Advanced > Features` and turn on the multichannel marketing feature.
2. Go to the Marketing > Overview page and confirm Pinterest for WooCommerce appears in the "Installed marketing extensions" section
3. Ensure you have **NOT** set up the extension or completed onboarding. If you have, reset the extension data.
4. Confirm that a link to "Finish setup" is displayed next to each extension name if it has not been set up and links to the onboarding/setup process.
5. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with a "Manage" link and points to the extension's settings page.
6. Confirm that the product sync / feed status is correct and it changes depending on the statuses displayed on the Products Catalog page of the extension
7. Confirm that the number of errors displayed is consistent with the number of errors in the Products Catalog page

For a compatibility check:

1. Use a WooCommerce version older than v7.4 (use 7.3, for example) - This means that the MCM feature classes will not be included in the WooCommerce code
2. Reload the marketing page and confirm that there are no errors loading the page

### Additional details:


### Changelog entry

>
